### PR TITLE
bpo-36345: Doc: make serve uses http.server instead of Tools/scripts/server.py 

### PR DESCRIPTION
**Pull request by matrixise**
https://bugs.python.org/issue36345

### CODE REVIEW

**Comment by matrixise** [comment]
@JulienPalard here is the PR

**Review by vstinner — CHANGES_REQUESTED** [review]
https://bugs.python.org/issue36345#msg338247

**Comment by matrixise** [comment]
I have made the requested changes; please review again

**Comment by vstinner** [comment]
https://bugs.python.org/issue36345#msg338637

**Comment by matrixise** [comment]
I have made the requested changes; please review again

**Comment by matrixise** [comment]
I added this PR https://github.com/python/cpython/pull/12511

**Comment by matrixise** [comment]
@vstinner I have updated this PR with your remark (modify the makefile first). this is the part-0 for this issue.

The part-2 is #12511 (sorry, I don't rename the branch)
The part-1 will arrive after you have approved the part-0.

I have made the requested changes; please review again

Thank you

File: Misc/NEWS.d/next/Documentation/2019-03-18-15-11-24.bpo-36345.aRmIHE.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Use :mod:`http.server` with ``-d`` parameter instead of
```
Comment (vstinner) [inline Misc/NEWS.d/next/Documentation/2019-03-18-15-11-24.bpo-36345.aRmIHE.rst]: What does "use http.server"? You should mention "make server" somewhere.

**Comment by matrixise** [comment]
Hi @vstinner thank you for your review

I have made the requested changes; please review again

**Comment by vstinner** [comment]
Please squash your commits into a single commit and update the PR title and description. It became very unclear what is the purpose of this change, because of its long history.

**Comment by hugovk** [comment]
`make serve` has since been replaced with `make htmllive`, which uses https://github.com/sphinx-doc/sphinx-autobuild:

* `make serve` was removed in https://github.com/python/cpython/issues/80510.
* `make htmllive` was added in https://github.com/python/cpython/pull/111900.

**Pull request closed without merging**